core/src/main/java/io/opencensus/stats/View.java

```diff
@@ -42,20 +48,43 @@
   public abstract Measure getMeasure();
 
   /**
+   * The {@link Aggregation}s associated with this {@link View}.
+   */
+  public abstract List<Aggregation> getAggregations();
+
+  /**
    * Columns (a.k.a Tag Keys) to match with the associated {@link Measure}.
-   * If no dimensions are specified, then all stats are recorded. Columns must be unique.
    *
-   * <p>Note: The returned list is unmodifiable, attempts to update it will throw an
-   * UnsupportedOperationException.
+   * <p>{@link Measure} will be recorded in a "greedy" way. That is, every view aggregates every
+   * measure. This is similar to doing a GROUPBY on view’s columns. Columns must be unique.
    */
   public abstract List<TagKey> getColumns();
 
   /**
-   * Applies the given match function to the underlying data type.
+   * Returns the time {@link Window} for this {@code View}.
+   *
+   * @return the time {@link Window}.
    */
-  public abstract <T> T match(
-      Function<? super DistributionView, T> p0,
-      Function<? super IntervalView, T> p1);
+  public abstract Window getWindow();
+
+  /**
+   * Constructs a new {@link View}.
+   */
+  public static View create(
+      Name name,
+      String description,
+      Measure measure,
+      List<Aggregation> aggregations,
```

Comment:
Make sense to me. Java does not have such an interface as OrderedSet, and SortedSet does not quite meet our need. I would prefer to keep columns and aggregations as lists and do verification on duplicates when creating views, so that users could rely on their order to match TagKey/TagValue and Aggregation/AggregationData. In addition, ViewData will contain only the "data" part (TagValue and AggregationData), which could be better for its on-the-wire representation.